homeassistant/components/folder_watcher.py

```diff
@@ -0,0 +1,85 @@
+"""
+Component for monitoring activity on a folder.
+"""
+import os
+import logging
+import voluptuous as vol
+from watchdog.observers import Observer
+from watchdog.events import FileSystemEventHandler
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['watchdog']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_FOLDERS = 'folders'
+DOMAIN = "folder_watcher"
+DEFAULT_IS_RECURSIVE = True
+EVENT_TYPE = "event_type"
+IS_DIRECTORY = "is_directory"
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_FOLDERS):
+            vol.All(cv.ensure_list, [cv.isdir]),
+        })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
```

Comment:
Now that you've converted this into a platform, you also need to change this to a `setup_platform` function. See this for an example:

https://github.com/home-assistant/home-assistant/blob/c462292e4d3017401a7def24ca8051b34ad605f1/homeassistant/components/sensor/folder.py#L45-L53

Reply:
Oh yes, I wrote that :)